### DISCUSSION TITLE
How does throttle cmd affect the velocity of F16?

### DISCUSSION THREAD

**Discussion by theorynice** [JSBSim-Team/jsbsim#971] [General]
How does throttle cmd affect the velocity of F16? In the f16 aircraft configuration files about the Drag axis, there is no aero coefficient related to propulsion. May I ignore something important?

**Comment by theorynice** [comment]
Does the force and moments produced by the engine are calculated independently? And then sum the force and moments generated by \<aerodynamics>~ \</aerodynamics> in configuration files as the final force and moments acting on the aircraft.

**Comment by seanmcleod** [comment]
There are a number of forces and moments that need to be calculated and applied, i.e.

https://github.com/JSBSim-Team/jsbsim/blob/ae4652a268d0475dbf93d47a32ab7ddf4120aaef/doc/JSBSimMainLoopFlow.dot#L61

**Reply by theorynice** [reply]
Thanks for your explanation helping me very much

**Comment by theorynice** [comment]
Thanks to all the guys in JSBSim community.  I have a paper in revised.  In the manuscript, I have claimed the used model is from JSBSim, but one reviewer hopes I can clarify the constraints and assumptions in F16 model.  The constraints may be the maximum speed and altitude, or performance limitations in various flight regimes. The assumptions may be the simplifications of each system component. For the above assumptions and constraints, are there any readily available references? If not, which serious constraints and assumptions can be listed in paper? Extremely grateful and thankful！

**Comment by seanmcleod** [comment]
Out of interest what is the subject of your paper?

You won't find any explicit constraints defined inside the FDM. For example, if you look inside `f16.xml` you won't find a specific constraint like `maximum ceiling = 60,000ft` which JSBSim somehow enforces. Rather JSBSim simply calculates the forces and moments from all the various systems, and uses the current mass and moment of inertia etc. to execute the EOM (Equations Of Motion) to calculate the present state for each timestep.

Now in theory if the input data to the FDM is accurate then you should end up with fairly accurate constraints in JSBSim that match the published physical constraints, but you would need to test them out yourself to see how accurate they are. For example in the `README` file in the F-16 directory the following from published sources is mentioned:

> Max speed:   Mach 2.05 at 39,000 Ft
Ceiling:     60,000 Ft

In terms of performance limitations here is an example of generating an Energy Maneuverability (EM) diagram using some Python code I wrote. So for example you could potentially take the required parameters, e.g. like maximum lift coefficient from the F-16 FDM etc. and generate an EM diagram. 

https://github.com/seanmcleod/EMDiagrams/blob/master/EMDiagrams.ipynb

However that only covers a small part of the overall performance limitations.

In terms of simplifications, sure there will be lots, for example there are much higher fidelity gas turbine models that others like say engine manufacturers use compared to the one implemented in JSBSim, but you won't find any real detailed documentation on the differences between the JSBSim model and higher fidelity models.

**Reply by theorynice** [reply]
Thanks for your patient explanation and kind suggestion. My paper title is "Autopilot Controller of Fixed-Wing Planes Based on Curriculum Reinforcement Learning Scheduled by Adaptive Learning Curve", which utilizes RL to design an autopilot controller that has a relatively higher performance. After considering, the response is below:
"
**Rigid body assumption:** JSBSim treats the F-16 as a rigid body, ignoring the effects of structural elasticity and deformation on flight dynamics.
**Ignoring Earth's curvature assumption:** JSBSim ignores the effect of Earth's curvature on flight dynamics when simulating the flight of the F-16.
**Fixed center of gravity assumption:** JSBSim assumes that the center of gravity of the F-16 is fixed and does not change with factors such as fuel consumption and weapon launch.
**Surface defection constraints:** The control surfaces have deflection limits. In F-16, the deflections of the elevator, aileron, and rudder are $\pm 25^{\circ}, \pm 21.5^{\circ}, \pm 30^{\circ}$ respectively.
**Thrust constraints:** The thrust of the engine has a maximum and a minimum value at different altitudes. For specific values, please see https://ntrs.nasa.gov/citations/19800005879. 
It should be noted that although JSBSim made the above assumptions and constraints when building the F-16 model, this does not mean that its simulation results are inaccurate or unreliable. On the contrary, JSBSim has been widely used and recognized in fields such as aviation, aerospace, and unmanned aerial vehicles, providing valuable reference for related research and applications.
"
I would appreciate it if you have any more suggestions!

**Comment by seanmcleod** [comment]
> **Rigid body assumption**: JSBSim treats the F-16 as a rigid body,

I would agree with this.

> **Ignoring Earth's curvature assumption**: JSBSim ignores the effect of Earth's curvature...

Not true, JSBSim doesn't assume a flat earth model.

> **Fixed center of gravity assumption**: JSBSim assumes that the center of gravity of the F-16 is fixed

Not true, JSBSim does model the changing mass based on fuel burn and also point mass changes, e.g. dropping a bomb etc. and it's change to the moments of inertia etc.

> **Surface defection constraints**: The control surfaces have deflection limits.

Sure, although I guess it depends what you mean by a constraint in this example. It's not more constrained than the physical aircraft, it's simply modelling the actual physical constraints of the real aircraft.

> **Thrust constraints**: The thrust of the engine has a maximum and a minimum value at different altitudes. For specific values, please see NASA report...

Why point at the NASA report rather than pointing to the actual JSBSim engine model that the JSBSim F-16 aircraft uses, i.e. 

https://github.com/JSBSim-Team/jsbsim/blob/master/engine/F100-PW-229.xml

In addition to taking density altitude into account it also takes Mach number into account.

**Reply by theorynice** [reply]
Thank you very much, I have modified the errors according to your kind correction. The constraints are well done. Could you supply some main assumptions that simplify the simulation in JSBSim?
